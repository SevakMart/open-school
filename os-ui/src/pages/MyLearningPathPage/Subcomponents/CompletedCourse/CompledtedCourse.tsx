import { useTranslation } from 'react-i18next';
import { UserCourseType } from '../../../../types/UserCourseType';
import Button from '../../../../component/Button/Button';
import styles from './CompletedCourse.module.scss';

const CompletedCourse = ({ title, courseStatus, grade }:Pick<UserCourseType, 'title'|'courseStatus'|'grade'>) => {
  const {
    separator, mainContainer, container, courseTitle,
    containerTitle, statusContent, gradeContent, buttonContainer,
  } = styles;
  const { t } = useTranslation();
  return (
    <div className={mainContainer}>
      <p data-testid={title} className={courseTitle}>{title}</p>
      <div className={separator} />
      <div className={container}>
        <p className={containerTitle}>{t('string.myLearningPaths.completedCourse.status')}</p>
        <p data-testid={courseStatus} className={statusContent}>{courseStatus}</p>
      </div>
      <div className={separator} />
      <div className={container}>
        <p className={containerTitle}>{t('string.myLearningPaths.completedCourse.grade')}</p>
        <p data-testid={grade} className={gradeContent}>
          {grade}
          /100
        </p>
      </div>
      <div className={separator} />
      <div className={buttonContainer}>
        <Button.MainButton className={['rateCourse']} onClick={() => null}>
          {t('string.myLearningPaths.completedCourse.rateCourse')}
        </Button.MainButton>
      </div>
    </div>
  );
};
export default CompletedCourse;
