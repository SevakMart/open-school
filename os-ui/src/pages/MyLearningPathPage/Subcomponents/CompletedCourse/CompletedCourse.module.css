.mainContainer {
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: none;
  background-color: #ffffff;
  border-radius: 20px;
  margin: 0 88px;
}
.mainContainer .courseTitle {
  font-family: "Helvetica Neue";
  font-style: normal;
  font-weight: 700;
  font-size: 24px;
  line-height: 29px;
  text-decoration-line: underline;
  color: #222427;
  margin: 54px 284px 54px 48px;
}
.mainContainer .container {
  margin: 43px 89px;
}
.mainContainer .container .containerTitle {
  font-family: "Montserrat";
  font-style: normal;
  font-weight: 500;
  font-size: 14px;
  line-height: 17px;
  color: #848A9D;
  margin: 0;
}
.mainContainer .container .gradeContent {
  font-family: "Montserrat";
  font-style: normal;
  font-weight: 500;
  font-size: 16px;
  line-height: 24px;
  color: #333941;
}
.mainContainer .container .statusContent {
  background-color: #D9DBE9;
  border-radius: 16px;
  font-family: "Montserrat";
  font-style: normal;
  font-weight: 500;
  font-size: 12px;
  line-height: 15px;
  color: #5E617B;
  padding: 0.344rem 1rem 0.344rem 1rem;
}
.mainContainer .buttonContainer {
  margin-left: 184px;
}
.mainContainer .separator {
  background-color: #EEEFF4;
  border-radius: 7px;
  width: 2px;
  height: 73px;
}/*# sourceMappingURL=CompletedCourse.module.css.map */