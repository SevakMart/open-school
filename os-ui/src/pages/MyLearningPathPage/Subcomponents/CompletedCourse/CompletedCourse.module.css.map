{"version":3,"sources":["CompletedCourse.module.scss","../../../../styles/Mixins/FlexMixins.scss","../../../../styles/Configs/Colors.scss","CompletedCourse.module.css","../../../../styles/Mixins/FontMixin.scss","../../../../styles/Configs/Fonts.scss"],"names":[],"mappings":"AAKA;ECJE,aAAA;EACA,mBDIqB;ECHrB,mBDGyB;ECFzB,qBDEgC;EAC9B,yBEHe;EFIf,mBAAA;EACA,cAAA;AGDJ;AHEI;EITF,6BAAA;EACA,kBJSyC;EIRzC,gBJQgD;EIPhD,eJOoD;EINpD,iBJMyD;EACnD,+BAAA;EACA,cEVY;EFWZ,4BAAA;AGIR;AHFI;EACI,iBAAA;AGIR;AHHQ;EIjBN,yBCDY;EDEZ,kBJiB2C;EIhB3C,gBJgBkD;EIflD,eJesD;EIdtD,iBJc2D;EACjD,cAAA;EACA,SAAA;AGSZ;AHPQ;EItBN,yBCDY;EDEZ,kBJsB2C;EIrB3C,gBJqBkD;EIpBlD,eJoBsD;EInBtD,iBJmB2D;EACjD,cEpBO;ACiCnB;AHXQ;EACI,yBAAA;EACA,mBAAA;EI5BV,yBCDY;EDEZ,kBJ4B2C;EI3B3C,gBJ2BkD;EI1BlD,eJ0BsD;EIzBtD,iBJyB2D;EACjD,cE7BM;EF8BN,oCAAA;AGiBZ;AHdI;EACI,kBAAA;AGgBR;AHdI;EACI,yBAAA;EACA,kBAAA;EACA,UAAA;EACA,YAAA;AGgBR","file":"CompletedCourse.module.css"}