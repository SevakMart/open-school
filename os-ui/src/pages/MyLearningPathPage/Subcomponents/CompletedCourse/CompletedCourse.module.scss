@import "../../../../styles/Mixins/FlexMixins.scss";
@import "../../../../styles/Mixins/FontMixin.scss";

.mainContainer{
    @include flexStyle(row,center,space-between);
    background-color: #FFFFFF;
    border-radius:1.25rem;
    width:100%;
    margin-bottom:7%;
    padding-left:3rem;
    padding-right:3rem;

    .container{
        @include flexStyle(column,center,space-between);
        .courseTitle{
            @include fontStyle('Helvetica Neue',normal,700,1.5rem,1.813rem);
            text-decoration-line: underline;
            color: #222427;
        }
        .containerTitle{
            @include fontStyle('Montserrat',normal,500,0.875rem,1.063rem);
            color: #848A9D;
        }
        .gradeContent{
            @include fontStyle('Montserrat',normal,500,1rem,1.5rem);
            color: #333941;
        }
        .statusContent{
            background-color: #D9DBE9;
            border-radius:1rem;
            @include fontStyle('Montserrat',normal,500,0.75rem,0.938rem);
            color: #5E617B;
            padding:0.344rem 1rem 0.344rem 1rem;
        }
        &:nth-child(1){
            width:25%;
            border:none;
        }
    }
    .buttonContainer button{
        border: 2px solid #848A9D;
        background-color: transparent;
        border-radius:1rem;
        @include fontStyle('Montserrat',normal,500,1rem,1.25rem);
        color:#848A9D;
        padding:1rem 2rem 1rem 2rem;
        cursor:pointer;
    }
    .separator{
        background-color:#EEEFF4;
        border-radius: 0.469rem;
        width:0.125rem;
        height:5rem;
    }
}