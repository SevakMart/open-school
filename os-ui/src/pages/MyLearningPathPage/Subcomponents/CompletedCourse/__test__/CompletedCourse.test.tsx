import { render, screen } from '@testing-library/react';
import CompletedCourse from '../CompledtedCourse';

describe('Create test cases for the Completed course component', () => {
  test('Create a snapshot test', () => {
    const { asFragment } = render(
      <CompletedCourse
        title="React js"
        courseStatus="COMPLETED"
        grade={90}
      />,
    );
    expect(asFragment()).toMatchSnapshot();
  });
  test('Check if the content is as expected', () => {
    render(
      <CompletedCourse
        title="React js"
        courseStatus="COMPLETED"
        grade={90}
      />,
    );
    const courseTitleElement = screen.queryByTestId('React js');
    const statusContentElement = screen.queryByTestId('COMPLETED');
    const gradeContentElement = screen.queryByTestId(90);

    expect(courseTitleElement).toBeInTheDocument();
    expect(courseTitleElement).toHaveTextContent('React js');
    expect(statusContentElement).toBeInTheDocument();
    expect(statusContentElement).toHaveTextContent('COMPLETED');
    expect(gradeContentElement).toBeInTheDocument();
    expect(gradeContentElement).toHaveTextContent('90/100');
  });
});
