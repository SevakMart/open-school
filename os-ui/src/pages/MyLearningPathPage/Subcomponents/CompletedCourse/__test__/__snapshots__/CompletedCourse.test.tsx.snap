// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create test cases for the Completed course component Create a snapshot test 1`] = `
<DocumentFragment>
  <div
    class="mainContainer"
  >
    <p
      class="courseTitle"
      data-testid="React js"
    >
      React js
    </p>
    <div
      class="separator"
    />
    <div
      class="container"
    >
      <p
        class="containerTitle"
      >
        string.myLearningPaths.completedCourse.status
      </p>
      <p
        class="statusContent"
        data-testid="COMPLETED"
      >
        COMPLETED
      </p>
    </div>
    <div
      class="separator"
    />
    <div
      class="container"
    >
      <p
        class="containerTitle"
      >
        string.myLearningPaths.completedCourse.grade
      </p>
      <p
        class="gradeContent"
        data-testid="90"
      >
        90/100
      </p>
    </div>
    <div
      class="separator"
    />
    <div
      class="buttonContainer"
    >
      <button
        class="rateCourse"
        type="button"
      >
        string.myLearningPaths.completedCourse.rateCourse
      </button>
    </div>
  </div>
</DocumentFragment>
`;
